### ISSUE TITLE
[lx] /proc/$pid/stat tty_nr is an untranslated illumos dev_t

### ISSUE THREAD

**Issue by nathan7** [joyent/illumos-joyent#100]
# Desired behaviour, observed on native Linux
```
[nathan@koala:~]$ uname -a
Linux koala 4.3.6 #1-NixOS SMP Fri Feb 19 22:35:24 UTC 2016 x86_64 GNU/Linux
[nathan@koala:~]$ awk '{print $7}' /proc/self/stat
34825
[nathan@koala:~]$ sudo -i
[sudo] password for nathan: 
[root@koala:~]# 
```

# Observed behaviour
```
[nathan@991490c2-bf90-479b-dfeb-fd9a723a9ba4:~]$ uname -a
Linux 991490c2-bf90-479b-dfeb-fd9a723a9ba4 3.13.0 BrandZ virtual linux x86_64 GNU/Linux
[nathan@991490c2-bf90-479b-dfeb-fd9a723a9ba4:~]$ awk '{print $7}' /proc/self/stat
3
[nathan@991490c2-bf90-479b-dfeb-fd9a723a9ba4:~]$ sudo -i
sudo: no tty present and no askpass program specified
[nathan@991490c2-bf90-479b-dfeb-fd9a723a9ba4:~]$ 
```

# Patched behaviour
```
[nathan@991490c2-bf90-479b-dfeb-fd9a723a9ba4:~]$ uname -a
Linux 991490c2-bf90-479b-dfeb-fd9a723a9ba4 3.13.0 BrandZ virtual linux x86_64 GNU/Linux
[nathan@991490c2-bf90-479b-dfeb-fd9a723a9ba4:~]$ awk '{print $7}' /proc/self/stat
34819
[nathan@991490c2-bf90-479b-dfeb-fd9a723a9ba4:~]$ sudo -i
[sudo] password for nathan: 
[root@991490c2-bf90-479b-dfeb-fd9a723a9ba4:~]# 
```

**Comment by jjelinek** [comment]
Thanks, I filed OS-5258 for this.
Jerry

**Comment by jjelinek** [comment]
File OS-5258 for this issue.

**Comment by nathan7** [comment]
@jjelinek I see you're reading this from email — unsure if that mentions [my patch](https://github.com/nathan7/illumos-joyent/commit/399d2c710762d4a798ae22a8f15559a5662160a5)

**Comment by jjelinek** [comment]
Nathan,

Thanks. While your patch is correct, I'd like to introduce a comprehensive
solution for this across all of the lx components that are mangling the
device number. I should have something ready shortly.

Thanks again,
Jerry

**Comment by nathan7** [comment]
Awesome! Looking forward to something cleaner than my hotfix.

**Comment by jjelinek** [comment]
Can you tell me what platform you are running. I am observing different behavior on the current build:

**Comment by jjelinek** [comment]
Thanks again, fixed with commit feb82da.
